src/main/java/com/redhat/vertx/pipeline/AbstractStep.java

```diff
@@ -20,78 +16,17 @@
     protected Engine engine;
     protected String name;
     protected Vertx vertx;
-    private JsonObject stepConfig;
-    private Duration timeout;
-    private String registerTo;
```

Comment:
Not sure I agree with taking all these out. If you extend Abstract Step, you really should only need write the execute method.